bus-mapping/src/circuit_input_builder.rs

```diff
@@ -286,6 +286,47 @@ impl<'a, C: CircuitsParams> CircuitInputBuilder<C> {
         }
     }
 
+    fn check_and_chunk(
+        &mut self,
+        geth_trace: &GethExecTrace,
+        tx: &'a mut Transaction,
+        tx_ctx: &'a mut TransactionContext,
+        i_step: Option<(usize, &GethExecStep)>,
```

Comment:
I'd suggest 
```diff
- i_step: Option<(usize, &GethExecStep)>,
+ geth_steps: Option<&[GethExecStep]>
```
and let caller bypassing `Some(&geth_trace.struct_logs[i..])` so we do not need extra `i` variable